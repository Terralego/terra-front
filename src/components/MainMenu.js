import React from 'react';
import {
  Classes,
  Navbar,
  NavbarGroup,
} from '@blueprintjs/core';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import withDeviceSize from '../hoc/withDeviceSize';
import { connectAuthProvider } from '../modules/Auth';

import MainMenuItem from './MainMenuItem';

export const MainMenu = ({
  className,
  navItems,
  ...props
}) => {
  if (!navItems.length) {
    return null;
  }

  return (
    <Navbar className={classnames(['navBar', Classes.DARK, className])} {...props}>
      {navItems.map((group, index) => (
        // eslint-disable-next-line react/no-array-index-key
        <NavbarGroup key={index} className="navBar__group">
          <ul>
            {group.map(item => (
              <li key={item.id}>
                <MainMenuItem {...item} />
              </li>
            ))}
          </ul>
        </NavbarGroup>
      ))}
    </Navbar>
  );
};

MainMenu.propTypes = {
  className: PropTypes.string,
  navItems: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        component: PropTypes.func,
        href: PropTypes.string,
        icon: PropTypes.oneOfType([PropTypes.string, PropTypes.shape({})]),
        id: PropTypes.string,
        label: PropTypes.string,
      }),
    ),
  ),
};

MainMenu.defaultProps = {
  className: '',
  navItems: [],
};

export default withDeviceSize()(connectAuthProvider('authenticated', 'logoutAction')(MainMenu));
