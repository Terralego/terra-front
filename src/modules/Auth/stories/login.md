LoginForm component will use a `translate` function prop to translate wordings with the followings keys:

| key | Default translation |
|-----|---------------------|
| 'auth.signupform.email.invalid' |  'Invalid email' |
| 'auth.signupform.email.help' | 'Type your email' |
| 'auth.signupform.email.label' | 'Email' |
| 'auth.signupform.email.info' | 'required' |
| 'auth.signupform.email.placeholder' | 'Email' |
| 'auth.signupform.password.invalid' | 'Invalid email' |
| 'auth.signupform.password.help' | 'Type your email' |
| 'auth.signupform.password.label' | 'Email' |
| 'auth.signupform.password.info' | 'required' |
| 'auth.signupform.password.placeholder' | 'Email' |
| 'auth.signupform.submit' |' Create account' |
