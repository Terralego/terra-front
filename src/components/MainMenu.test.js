import React from 'react';
import renderer from 'react-test-renderer';

import { MainMenu } from './MainMenu';
import login from '../images/log-in.svg';
import logout from '../images/log-out.svg';
import infosign from '../images/info-sign.svg';

jest.mock('./NavBarItemDesktop', () => () => 'NavBarItemDesktop');
jest.mock('./NavBarItemTablet', () => () => 'NavBarItemTablet');

it('should render properly', () => {
  const tree = renderer.create(
    <MainMenu
      navItems={[[
        { label: 'log in', icon: login },
        { label: 'log out', icon: logout },
        { label: 'info sign', icon: infosign },
      ]]}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render properly on mobile size', () => {
  const tree = renderer.create(
    <MainMenu
      isMobileSized
      navItems={[[
        { label: 'log in', icon: login },
        { label: 'log out', icon: logout },
        { label: 'info sign', icon: infosign },
      ]]}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render properly with custom component', () => {
  const tree = renderer.create(
    <MainMenu
      navItems={[[
        { component: () => <button type="button">click me</button> },
        { component: () => <img src={login} alt="login logo" /> },
        { component: () => <div>check me<input type="checkbox" /></div> },
        { component: () => <div>search me<input type="text" /></div> },
      ]]}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
