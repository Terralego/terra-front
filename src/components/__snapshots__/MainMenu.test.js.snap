// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<div
  className="bp3-navbar navBar bp3-dark"
>
  <div
    className="bp3-navbar-group bp3-align-left navBar__group"
  >
    <ul>
      <li>
        NavBarItemDesktop
      </li>
      <li>
        NavBarItemDesktop
      </li>
      <li>
        NavBarItemDesktop
      </li>
    </ul>
  </div>
</div>
`;

exports[`should render properly with custom component 1`] = `
<div
  className="bp3-navbar navBar bp3-dark"
>
  <div
    className="bp3-navbar-group bp3-align-left navBar__group"
  >
    <ul>
      <li>
        <button
          type="button"
        >
          click me
        </button>
      </li>
      <li>
        <img
          alt="login logo"
          src="log-in.svg"
        />
      </li>
      <li>
        <div>
          check me
          <input
            type="checkbox"
          />
        </div>
      </li>
      <li>
        <div>
          search me
          <input
            type="text"
          />
        </div>
      </li>
    </ul>
  </div>
</div>
`;
